rpc/client/client_test.go

```diff
@@ -214,6 +215,77 @@ func TestGetBlock(t *testing.T) {
 	require.NoError(err)
 }
 
+func TestGetCommit(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+	mockApp, rpc := getRPC(t)
+	mockApp.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	mockApp.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+
+	block := getRandomBlock(1, 5)
+	err := rpc.node.Start()
+	require.NoError(err)
+
+	err = rpc.node.Store.SaveBlock(block, &types.Commit{Height: block.Header.Height})
+	require.NoError(err)
+
+	commit, err := rpc.Commit(context.Background(), nil)
+	require.NoError(err)
+	require.NotNil(commit)
+	assert.Equal(block.Header.Height, uint64(commit.Height))
+}
+
+func TestBlockSearch(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+	mockApp, rpc := getRPC(t)
+	mockApp.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	mockApp.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+
+	heights := []int64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+	indexBlocks(t, rpc, heights)
+
+	tests := map[string]struct {
+		query      string
+		page       int
+		perPage    int
+		totalCount int
+		orderBy    string
+	}{
+		"block.height >= 1 AND end_event.foo <= 5": {
+			query:      "block.height >= 1 AND end_event.foo <= 5",
+			page:       1,
+			perPage:    5,
+			totalCount: 5,
+			orderBy:    "asc",
+		},
+		"block.height >= 2 AND end_event.foo <= 10": {
+			query:      "block.height >= 2 AND end_event.foo <= 10",
+			page:       1,
+			perPage:    3,
+			totalCount: 9,
+			orderBy:    "desc",
+		},
+		"begin_event.proposer = 'FCAA001' AND end_event.foo <= 5": {
+			query:      "begin_event.proposer = 'FCAA001' AND end_event.foo <= 5",
+			page:       1,
+			perPage:    5,
+			totalCount: 5,
+			orderBy:    "asc",
+		},
+	}
+
+	for name, test := range tests {
+		test := test
+		t.Run(name, func(t *testing.T) {
```

Comment:
You can use simple slice of structs instead of map, and query as a name. This will remove some repetition and fragility (now, if you change query, you need to remember to also change name).

Reply:
Yeah can do that, though I've seen quite some codebases follow the map approach for tests.